src/test/java/com/ipiecoles/java/java350/service/EmployeServiceTest.java

```diff
@@ -141,4 +143,131 @@ public void testEmbaucheEmployeManagerMiTempsMaster99999(){
+    }
+
+    @Test
+    public void testCalculPerformanceCommercialObjectifCaIsNegative(){
+        //Given
+        String matricule = "C00001";
+        Long caTraite = 15L;
+        Long objectifCa = -15L;
+
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals("L'objectif de chiffre d'affaire ne peut être négatif ou null !", e.getMessage());
+    }
+
+    //Test parametre pour performance attendu
+    //Si cA > objCa * 0.8 && cA < objCa * 0.95
+    //Si cA > objCa * 0.95 && cA < objCa * 1.05
+    //Si cA < objCa * 1.2 && cA > objCa * 1.05
+    //Si cA > objCa * 1.2
+    //Si cA > objCa * 1.2 + bonus perf
+    //Si cA > objCa * 0.8 && cA < objCa * 0.95 et perf haute
+    @ParameterizedTest(name = "performance {0}, matricule {1}, caTraite {2}, objectifCa {3}, performanceAttendu {4}")
+    @CsvSource({
+            "1, 'C00001', 1000, 1100, 1",
+            "1, 'C00001', 1000, 1050, 1",
+            "1, 'C00001', 1000, 950, 3",
+            "1, 'C00001', 1000, 0, 6",
+            "3, 'C00001', 1300, 1000, 8",
+            "8, 'C00001', 800, 1000, 7"
+    })
+    public void testCalculPerformanceCommercialAttendu(Integer performance, String matricule, Long caTraite, Long objectifCa, Integer performanceAttendu) throws EmployeException {
+        //Given
+        Employe employe = new Employe("Tako", "Yaki", matricule, LocalDate.now(), Entreprise.SALAIRE_BASE, performance, 1.0);
+        Mockito.when(employeRepository.findByMatricule(matricule)).thenReturn(employe);
+        Mockito.when(employeRepository.avgPerformanceWhereMatriculeStartsWith("C")).thenReturn(1.0);
+        when(employeRepository.save(Mockito.any(Employe.class))).thenAnswer(AdditionalAnswers.returnsFirstArg());
+
+        //When
+        employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa);
+
+        //Then
+        ArgumentCaptor<Employe> employeArgumentCaptor = ArgumentCaptor.forClass(Employe.class);
+        verify(employeRepository, times(1)).save(employeArgumentCaptor.capture());
+        Assertions.assertEquals(performanceAttendu, employeArgumentCaptor.getValue().getPerformance());
+    }
+
+    @Test
+    public void testCalculPerformanceCommercialIfEmployeIsNull() throws EmployeException {
+        //Given
+        String matricule = "C00001";
+        Long caTraite = 15L;
+        Long objectifCa = 15L;
+        Mockito.when(employeRepository.findByMatricule(matricule)).thenReturn(null);
+        //When
+        //Then
+        EmployeException e = Assertions.assertThrows(EmployeException.class, () -> employeService.calculPerformanceCommercial(matricule, caTraite, objectifCa));
+        Assertions.assertEquals(e.getMessage(), "Le matricule " + matricule + " n'existe pas !");
```

Comment:
Préférer **AssertJ** par rapport à **Junit**

```suggestion
        Assertions.assertThat(e.getMessage()).isEqualTo("Le matricule " + matricule + " n'existe pas !");
```